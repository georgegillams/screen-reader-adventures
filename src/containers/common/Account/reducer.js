import produce from 'immer';

import { logout, requestVerificationEmail } from './actions';

export const initialState = {
  loggingOut: false,
  logOutError: null,
  logOutResult: null,
  requestingVerificationEmail: false,
  requestVerificationEmailError: null,
  requestVerificationEmailResult: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case logout.REQUEST:
        draft.loggingOut = true;
        draft.logOutError = null;
        break;

      case logout.SUCCESS:
        draft.loggingOut = false;
        draft.logOutResult = payload;
        break;

      case logout.FAILURE:
        draft.loggingOut = false;
        draft.logOutError = payload;
        break;

      case requestVerificationEmail.REQUEST:
        draft.requestingVerificationEmail = true;
        draft.requestVerificationEmailError = null;
        break;

      case requestVerificationEmail.SUCCESS:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailResult = payload;
        break;

      case requestVerificationEmail.FAILURE:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailError = payload;
        break;
    }
  });

export default reducer;
