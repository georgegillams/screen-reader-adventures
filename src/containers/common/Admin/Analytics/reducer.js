import produce from 'immer';

import { load } from './actions';

export const initialState = {
  analytics: null,
  loading: false,
  loadError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.analytics = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;
    }
  });

export default reducer;
