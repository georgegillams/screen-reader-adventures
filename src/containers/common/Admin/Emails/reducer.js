import produce from 'immer';

import { load, resend } from './actions';

export const initialState = {
  emails: null,
  loading: false,
  loadError: null,

  emailToResend: null,
  resending: false,
  resendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.emails = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case resend.TRIGGER:
        draft.emailToResend = payload.emailToResend;
        break;

      case resend.REQUEST:
        draft.resending = true;
        draft.resendError = null;
        break;

      case resend.SUCCESS:
        draft.resending = false;
        break;

      case resend.FAILURE:
        draft.resending = false;
        draft.resendError = payload;
        break;
    }
  });

export default reducer;
