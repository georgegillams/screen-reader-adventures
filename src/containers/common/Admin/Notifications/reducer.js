import produce from 'immer';

import { load, create, remove } from './actions';

export const initialState = {
  notifications: null,
  loading: false,
  loadError: null,

  notificationToCreate: null,
  creating: false,
  createError: null,

  notificationToRemove: null,
  removing: false,
  removeError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.notifications = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case create.TRIGGER:
        draft.notificationToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case remove.TRIGGER:
        draft.notificationToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;
    }
  });

export default reducer;
