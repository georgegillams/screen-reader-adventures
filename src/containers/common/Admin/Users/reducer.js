import produce from 'immer';

import { load, remove, requestMagicLink, create, update } from './actions';

export const initialState = {
  users: null,
  loading: false,
  loadError: null,

  userToRemove: null,
  removing: false,
  removeError: null,

  userToLogIn: null,
  requesting: false,
  requestError: null,

  userToCreate: null,
  creating: false,
  createError: null,

  userToUpdate: null,
  updating: false,
  updateError: null,
  onUpdateSuccessCb: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.users = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case remove.TRIGGER:
        draft.userToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;

      case requestMagicLink.TRIGGER:
        draft.userToLogIn = payload;
        break;

      case requestMagicLink.REQUEST:
        draft.requesting = true;
        draft.requestError = null;
        break;

      case requestMagicLink.SUCCESS:
        draft.requesting = false;
        break;

      case requestMagicLink.FAILURE:
        draft.requesting = false;
        draft.requestError = payload;
        break;

      case create.TRIGGER:
        draft.userToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case update.TRIGGER:
        draft.userToUpdate = payload.userToUpdate;
        draft.onUpdateSuccessCb = payload.onUpdateSuccessCb;
        break;

      case update.REQUEST:
        draft.updating = true;
        draft.updateError = null;
        break;

      case update.SUCCESS:
        draft.updating = false;
        break;

      case update.FAILURE:
        draft.updating = false;
        draft.updateError = payload;
        break;
    }
  });

export default reducer;
