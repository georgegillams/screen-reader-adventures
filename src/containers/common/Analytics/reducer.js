import produce from 'immer';

import { sendAnalytic } from './actions';

export const initialState = {
  sending: false,
  sendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case sendAnalytic.TRIGGER:
        draft.analytic = payload;
        break;

      case sendAnalytic.REQUEST:
        draft.sending = true;
        draft.sendError = null;
        break;

      case sendAnalytic.SUCCESS:
        draft.sending = false;
        break;

      case sendAnalytic.FAILURE:
        draft.sending = false;
        draft.sendError = payload;
        break;
    }
  });

export default reducer;
