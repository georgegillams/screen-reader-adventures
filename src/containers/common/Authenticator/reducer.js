import produce from 'immer';
import cookie from 'react-cookies';

import { SESSION_COOKIE_KEY, COOKIE_EXPIRY_TIME } from 'helpers/storageConstants';
import { useLocalhost } from 'helpers/appConfig';

import { setUser, loadAuth } from './actions';

export const initialState = {
  loadingAuth: false,
  loadAuthError: null,
  user: undefined,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case setUser.TRIGGER:
        draft.user = payload;
        if (payload && payload.session) {
          cookie.save(SESSION_COOKIE_KEY, payload.session, {
            path: '/',
            secure: !useLocalhost,
            expires: new Date(Date.now() + COOKIE_EXPIRY_TIME),
          });
        }
        break;

      case loadAuth.REQUEST:
        draft.loadingAuth = true;
        draft.loadAuthError = null;
        break;

      case loadAuth.SUCCESS:
        draft.loadingAuth = false;
        draft.user = payload.user;
        break;

      case loadAuth.FAILURE:
        draft.loadingAuth = false;
        draft.loadAuthError = payload;
        break;
    }
  });

export default reducer;
