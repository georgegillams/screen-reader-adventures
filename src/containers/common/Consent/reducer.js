import produce from 'immer';

import { consent, deferConsent, setConsentReason, resetConsent } from './actions';
import { CONSENT_STATE_UNSET, CONSENT_STATE_ALLOWED, CONSENT_STATE_DEFERRED } from './constants';

export const initialState = {
  cookieConsent: null,
  cookieConsentReason: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case resetConsent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_UNSET;
        break;

      case consent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_ALLOWED;
        break;

      case deferConsent.TRIGGER:
        draft.cookieConsent = CONSENT_STATE_DEFERRED;
        break;

      case setConsentReason.TRIGGER:
        draft.cookieConsentReason = payload;
        break;
    }
  });

export default reducer;
