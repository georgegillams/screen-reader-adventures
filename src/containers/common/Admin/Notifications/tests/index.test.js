import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initialState } from '../reducer';
import { initialState as initialAuthenticatorState } from 'containers/common/Authenticator/reducer';

import configureStore from 'client-utils/common/redux/configure-store';

import AdminNotificationsIndex from '../index';
import AdminNotifications from '../Container';

describe('<AdminNotifications />', () => {
  let store;
  const spy = jest.fn();

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotificationsIndex />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with initial state', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser' },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with admin user', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loading=true', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
            loading: true,
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loadError', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
            loadError: { error: 'some_error', errorMessage: 'Something went wrong' },
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with notifications', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminNotifications
          load={spy}
          resend={spy}
          notificationsState={{
            ...initialState,
            notifications: [
              { id: '1', message: 'message1', type: 'success' },
              { id: '2', message: 'message2', type: 'warn' },
              { id: '3', message: 'message3', type: 'error' },
            ],
          }}
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'testUser', admin: true },
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
