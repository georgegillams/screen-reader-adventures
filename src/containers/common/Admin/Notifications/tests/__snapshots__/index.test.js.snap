// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminNotifications /> should render correctly - index 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
  </div>
</div>
`;

exports[`<AdminNotifications /> should render correctly with admin user 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin notifications
    </h1>
  </div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="message_FORM_ID"
    >
      Message
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Message"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="message_FORM_ID"
        name="Message"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="type_FORM_ID"
    >
      Type
    </label>
    <div
      class="forms__component forms__component__text-box"
      placeholder="Type"
    >
      <div
        class="input__outer"
      >
        <select
          aria-disabled="false"
          aria-invalid="false"
          class="input__inner input__inner--select"
          id="type_FORM_ID"
          name="Type"
        >
          <option
            value="success"
          >
            Success (green)
          </option>
          <option
            value="warn"
          >
            Warning (orange)
          </option>
          <option
            value="error"
          >
            Error (red)
          </option>
        </select>
        <svg
          class="icon input__icon input__icon--valid"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
    </div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Create notification
      </span>
    </button>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-notifications__control"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload notifications
      </span>
    </button>
  </div>
</div>
`;

exports[`<AdminNotifications /> should render correctly with initial state 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="skeleton__outer skeleton__section"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
    <div
      class="skeleton__outer skeleton__card"
      style="margin-top: 1.2rem;"
    >
      <div
        class="skeleton__shimmer"
        style="margin-left: -0px;"
      />
    </div>
  </div>
</div>
`;

exports[`<AdminNotifications /> should render correctly with loadError 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin notifications
    </h1>
  </div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="message_FORM_ID"
    >
      Message
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Message"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="message_FORM_ID"
        name="Message"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="type_FORM_ID"
    >
      Type
    </label>
    <div
      class="forms__component forms__component__text-box"
      placeholder="Type"
    >
      <div
        class="input__outer"
      >
        <select
          aria-disabled="false"
          aria-invalid="false"
          class="input__inner input__inner--select"
          id="type_FORM_ID"
          name="Type"
        >
          <option
            value="success"
          >
            Success (green)
          </option>
          <option
            value="warn"
          >
            Warning (orange)
          </option>
          <option
            value="error"
          >
            Error (red)
          </option>
        </select>
        <svg
          class="icon input__icon input__icon--valid"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
    </div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Create notification
      </span>
    </button>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-notifications__control"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload notifications
      </span>
    </button>
  </div>
  <span
    class="paragraph"
  >
    <span
      class="error"
    >
      Could not load notifications
      <br />
      Something went wrong
    </span>
  </span>
</div>
`;

exports[`<AdminNotifications /> should render correctly with loading=true 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin notifications
    </h1>
  </div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="message_FORM_ID"
    >
      Message
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Message"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="message_FORM_ID"
        name="Message"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="type_FORM_ID"
    >
      Type
    </label>
    <div
      class="forms__component forms__component__text-box"
      placeholder="Type"
    >
      <div
        class="input__outer"
      >
        <select
          aria-disabled="false"
          aria-invalid="false"
          class="input__inner input__inner--select"
          id="type_FORM_ID"
          name="Type"
        >
          <option
            value="success"
          >
            Success (green)
          </option>
          <option
            value="warn"
          >
            Warning (orange)
          </option>
          <option
            value="error"
          >
            Error (red)
          </option>
        </select>
        <svg
          class="icon input__icon input__icon--valid"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
    </div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Create notification
      </span>
    </button>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy button__outer--disabled admin-notifications__control"
      disabled=""
      type="button"
    >
      <span
        class="button__child button__child--loading"
      >
        Reload notifications
      </span>
      <div
        aria-hidden="true"
        class="button__loading-spinner"
      >
        <div
          class="spinner__outer"
        >
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
          <div
            class="spinner__inner"
          />
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`<AdminNotifications /> should render correctly with notifications 1`] = `
<div>
  <div>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline page-title__link"
      href="/admin"
    >
      ⇠ Admin
    </a>
    <h1
      class="page-title__heading page-title__heading--with-link"
    >
      Admin notifications
    </h1>
  </div>
  <div
    class=""
  >
    <label
      class="forms__component forms__component__label"
      for="message_FORM_ID"
    >
      Message
    </label>
    <div
      class="input__outer forms__component forms__component__text-box"
      placeholder="Message"
    >
      <input
        aria-disabled="false"
        aria-invalid="false"
        class="input__inner"
        id="message_FORM_ID"
        name="Message"
        value=""
      />
      <svg
        class="icon input__icon input__icon--valid"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
        />
      </svg>
    </div>
    <label
      class="forms__component forms__component__label"
      for="type_FORM_ID"
    >
      Type
    </label>
    <div
      class="forms__component forms__component__text-box"
      placeholder="Type"
    >
      <div
        class="input__outer"
      >
        <select
          aria-disabled="false"
          aria-invalid="false"
          class="input__inner input__inner--select"
          id="type_FORM_ID"
          name="Type"
        >
          <option
            value="success"
          >
            Success (green)
          </option>
          <option
            value="warn"
          >
            Warning (orange)
          </option>
          <option
            value="error"
          >
            Error (red)
          </option>
        </select>
        <svg
          class="icon input__icon input__icon--valid"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
    </div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy forms__component forms__component__button"
      type="button"
    >
      <span
        class="button__child"
      >
        Create notification
      </span>
    </button>
  </div>
  <div>
    <button
      class="button__outer button__outer--regular button__outer--no-bouncy admin-notifications__control"
      type="button"
    >
      <span
        class="button__child"
      >
        Reload notifications
      </span>
    </button>
  </div>
  <div
    class="admin-notifications__entity"
  >
    <div
      class="notification-comp__notification notification-comp__notification--success"
    >
      message1
    </div>
    <button
      class="button__outer button__outer--destructive button__outer--no-bouncy button__outer--no-bouncy--destructive"
      type="button"
    >
      <span
        class="button__child"
      >
        Remove
      </span>
    </button>
  </div>
  <div
    class="admin-notifications__entity"
  >
    <div
      class="notification-comp__notification notification-comp__notification--warn"
    >
      message2
    </div>
    <button
      class="button__outer button__outer--destructive button__outer--no-bouncy button__outer--no-bouncy--destructive"
      type="button"
    >
      <span
        class="button__child"
      >
        Remove
      </span>
    </button>
  </div>
  <div
    class="admin-notifications__entity"
  >
    <div
      class="notification-comp__notification notification-comp__notification--error"
    >
      message3
    </div>
    <button
      class="button__outer button__outer--destructive button__outer--no-bouncy button__outer--no-bouncy--destructive"
      type="button"
    >
      <span
        class="button__child"
      >
        Remove
      </span>
    </button>
  </div>
</div>
`;

exports[`<AdminNotifications /> should render correctly with user 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="admin-only"
    >
      Admin only
    </h2>
    <span
      class="paragraph"
    >
      You need to be logged in with an admin account to view this content.
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/login"
      >
        Got a different admin account? Log in here.
      </a>
    </span>
  </div>
</div>
`;
